### ISSUE TITLE
"Already continuation registered, can not await twice" - how to debug?

### ISSUE THREAD

**Issue by GainDeveloper** [Cysharp/UniTask#202]
Hello!
Apologies if this is a basic question - I've been struggling with this for a few days.
I'm getting the exception on the following await,

System.InvalidOperationException: Already continuation registered, can not await twice or get Status after await.

```csharp
var handle = Addressables.LoadAssetAsync<UnityEngine.Object>(args.RefCounter.Resource);
await handle;
```

As far as I can see there's no reason for this error to be thrown? I don't await the handle twice, so I am unsure how to properly debug this issue? Could it be caused elsewhere in the code since the tasks are pooled? The error isn't thrown on the first time this code runs - it's after many loads and under specific circumstances.
Any help on how to best track this down would be appreciated.

**Comment by GainDeveloper** [comment]
I ended up changing the above code to this which resolved the error. 
I'm assuming that there's something wrong with UniTasks handling of AsyncOperationHandle that was causing this.

```
var handle = Addressables.LoadAssetAsync<UnityEngine.Object>(args.RefCounter.Resource);
await UniTask.WaitUntil(() => { return handle.IsDone; });
```

**Comment by neuecc** [comment]
need error's stacktrace.

**Comment by AwaharaFujihiro** [comment]
Same error was thrown on this code.
Is this a same issue?

```C#
private static async UniTask TestAsync(CancellationToken ct)
{
    Debug.Log("TestAsync Start.");
    var count = 0;
    while (!ct.IsCancellationRequested)
    {
        try
        {
            Debug.Log($"TestAsync try count:{++count}");
            var task1 = new WaitUntil(() => Random.Range(0, 10) == 0).ToUniTask();
            var task2 = new WaitUntil(() => Random.Range(0, 10) == 0).ToUniTask();
            var task3 = new WaitUntil(() => Random.Range(0, 10) == 0).ToUniTask();
            await UniTask.WhenAny(task1, task2, task3);
        }
        catch (Exception e)
        {
            Debug.LogError(e);
            return;
        }
    }
    Debug.Log("TestAsync Finished.");
}
```

**Comment by neuecc** [comment]
Thank you.
I can confirm that, indeed, the pooling mechanism seems to be broken somehow.
I'll fix it.

**Comment by neuecc** [comment]
when Coroutine(Enumerator) returns immediately, will break pool.
Maybe it is not same as first reported, but actualy it is crtical bug.

**Comment by neuecc** [comment]
I've released `2.1.1`, it includes fix about IEnumerator.ToUniTask.
